packages/util-user-agent-node/src/index.spec.ts

```diff
@@ -54,4 +54,14 @@ describe("defaultUserAgent", () => {
     const userAgent = await defaultUserAgent({ serviceId: "s3", clientVersion: "0.1.0" })();
     expect(userAgent).toContainEqual([`app/${appId}`]);
   });
+
+  it("should memoize app id", async () => {
+    mockAppIdLoader.mockClear();
+    const appId = "appId12345";
+    mockAppIdLoader.mockResolvedValue(appId);
+    const userAgnetProvider = defaultUserAgent({ serviceId: "s3", clientVersion: "0.1.0" });
+    await userAgnetProvider();
+    await userAgnetProvider();
```

Comment:
Can you add a test that this returns the same array every time and if possible, has the expected contents or only contains `app/${appId}` once?